src/cript/nodes/core.py

```diff
@@ -242,6 +242,164 @@ def json(self):
+        Generates a long-form JSON representation of the current node and its hierarchy.
+
+        The long-form JSON includes complete details of the node, eliminating the need for
+         references to UUIDs to nodes stored in the CRIPT database. This comprehensive representation
+         is useful for offline storage of CRIPT nodes, transferring nodes between different CRIPT instances,
+         or for backup purposes.
+
+        The generated long-form JSON can be reloaded into the SDK using
+        [`cript.load_nodes_from_json()`](../../../utility_functions/#cript.nodes.util.load_nodes_from_json),
+        ensuring consistency and completeness of the node data.
+        However, it's important to note that this long-form JSON might not comply directly with the JSON schema
+        required for POST or PATCH requests to the CRIPT API.
+
+        Optional keyword arguments (`kwargs`) are supported and are passed directly to `json.dumps()`.
+        These arguments allow customization of the JSON output, such as formatting for readability
+        or pretty printing.
+
+        Parameters
+        ----------
+        **kwargs : dict, optional
+            Additional keyword arguments for `json.dumps()` to customize the JSON output, such as `indent`
+            for pretty-printing.
+
+        Returns
+        -------
+        str
+            A comprehensive JSON string representing the current node and its entire hierarchy in long-form.
+
+        Notes
+        -----
+        The `get_expanded_json()` method differs from the standard [`json`](./#cript.nodes.core.BaseNode.json)
+        property or method, which might provide a more condensed version of the node's data.
+
+        > For more information on condensed JSON and deserialization, please feel free to reference our discussion
+        > on [deserializing Python nodes to JSON](https://github.com/C-Accel-CRIPT/Python-SDK/discussions/177)
+
+        Examples
+        --------
+        >>> import cript
+        >>> # ============= Create all needed nodes =============
+        >>> my_project = cript.Project(name=f"my_Project")
+        >>> my_collection = cript.Collection(name="my collection")
+        >>> my_material_1 = cript.Material(
+        ...     name="my material 1", identifier=[{"bigsmiles": "my material 1 bigsmiles"}]
+        ... )
+        >>> my_material_2 = cript.Material(
+        ...     name="my material 2", identifier=[{"bigsmiles": "my material 2 bigsmiles"}]
+        ... )
+        >>> my_inventory = cript.Inventory(
+        ...     name="my inventory", material=[my_material_1, my_material_2]
+        ... )
+        >>> #  ============= Assemble nodes =============
+        >>> my_project.collection = [my_collection]
+        >>> my_project.collection[0].inventory = [my_inventory]
+        >>> #  ============= Get long form JSON =============
+        >>> long_form_json = my_project.get_expanded_json(indent=4)
+
+        ???+ info "Short JSON VS Long JSON"
```

Comment:
We should have called this section `Condensed JSON VS Expanded JSON`, but it is okay for now and we can come back to it later

Reply:
agreed